RecoParticleFlow/PFClusterProducer/src/PFClusterEMEnergyCorrector.cc

```diff
@@ -7,33 +13,218 @@ namespace {
+  _recHitsEE = cc.consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("recHitsEELabel"));
+  _vertices  = cc.consumes<reco::VertexCollection>(conf.getParameter<edm::InputTag>("verticesLabel"));
+  
+  autoDetectBunchSpacing_ = conf.getParameter<bool>("autoDetectBunchSpacing");
+
+  if (autoDetectBunchSpacing_) {
+    bunchSpacing_ = cc.consumes<int>(edm::InputTag("addPileupInfo","bunchSpacing"));
+    bunchSpacingManual_ = 0;
+  }
+  else {
+    bunchSpacingManual_ = conf.getParameter<int>("bunchSpacing");
+  }
+  
+  _condnames_mean_50ns.push_back("GBRForestD_ecalPFClusterCor_EB_pfSize1_mean_50ns");
+  _condnames_mean_50ns.push_back("GBRForestD_ecalPFClusterCor_EB_pfSize2_mean_50ns");
+  _condnames_mean_50ns.push_back("GBRForestD_ecalPFClusterCor_EB_pfSize3_mean_50ns");
+  _condnames_mean_50ns.push_back("GBRForestD_ecalPFClusterCor_EE_pfSize1_mean_50ns");
+  _condnames_mean_50ns.push_back("GBRForestD_ecalPFClusterCor_EE_pfSize2_mean_50ns");
+  _condnames_mean_50ns.push_back("GBRForestD_ecalPFClusterCor_EE_pfSize3_mean_50ns");
+
+  _condnames_sigma_50ns.push_back("GBRForestD_ecalPFClusterCor_EB_pfSize1_sigma_50ns");
+  _condnames_sigma_50ns.push_back("GBRForestD_ecalPFClusterCor_EB_pfSize2_sigma_50ns");
+  _condnames_sigma_50ns.push_back("GBRForestD_ecalPFClusterCor_EB_pfSize3_sigma_50ns");
+  _condnames_sigma_50ns.push_back("GBRForestD_ecalPFClusterCor_EE_pfSize1_sigma_50ns");
+  _condnames_sigma_50ns.push_back("GBRForestD_ecalPFClusterCor_EE_pfSize2_sigma_50ns");
+  _condnames_sigma_50ns.push_back("GBRForestD_ecalPFClusterCor_EE_pfSize3_sigma_50ns");
+  
+  _condnames_mean_25ns.push_back("GBRForestD_ecalPFClusterCor_EB_pfSize1_mean_25ns");
+  _condnames_mean_25ns.push_back("GBRForestD_ecalPFClusterCor_EB_pfSize2_mean_25ns");
+  _condnames_mean_25ns.push_back("GBRForestD_ecalPFClusterCor_EB_pfSize3_mean_25ns");
+  _condnames_mean_25ns.push_back("GBRForestD_ecalPFClusterCor_EE_pfSize1_mean_25ns");
+  _condnames_mean_25ns.push_back("GBRForestD_ecalPFClusterCor_EE_pfSize2_mean_25ns");
+  _condnames_mean_25ns.push_back("GBRForestD_ecalPFClusterCor_EE_pfSize3_mean_25ns");
+
+  _condnames_sigma_25ns.push_back("GBRForestD_ecalPFClusterCor_EB_pfSize1_sigma_25ns");
+  _condnames_sigma_25ns.push_back("GBRForestD_ecalPFClusterCor_EB_pfSize2_sigma_25ns");
+  _condnames_sigma_25ns.push_back("GBRForestD_ecalPFClusterCor_EB_pfSize3_sigma_25ns");
+  _condnames_sigma_25ns.push_back("GBRForestD_ecalPFClusterCor_EE_pfSize1_sigma_25ns");
+  _condnames_sigma_25ns.push_back("GBRForestD_ecalPFClusterCor_EE_pfSize2_sigma_25ns");
+  _condnames_sigma_25ns.push_back("GBRForestD_ecalPFClusterCor_EE_pfSize3_sigma_25ns");  
+  
+}
+
+
+void PFClusterEMEnergyCorrector::correctEnergies(const edm::Event &evt, const edm::EventSetup &es, const reco::PFCluster::EEtoPSAssociation &assoc, reco::PFClusterCollection& cs) {
+
+  int bunchspacing = 450;  
+  
+  if (autoDetectBunchSpacing_) {
```

Comment:
My hope is that making this increasingly horrible leads to a proper solution being implemented sooner...